src/free-style.ts

```diff
@@ -170,10 +170,11 @@ function stringifyProperties (properties: Properties) {
 
   for (const [name, value] of properties) {
     if (value != null) {
-      if (Array.isArray(value)) {
-        result.push(value.filter(x => x != null).map(x => styleToString(name, x)).join(';'))
+      const value2 = typeof value === 'function' ? value() : value
```

Comment:
Can you help me understand what would be broken and how?   The expectation with this function is that it is only ever called once.